### PR TITLE
Adding display parts to definition items to support FindAllReferences

### LINKED ISSUE #10541: Find all References should include symbol display parts for the definition item
Include symbol display parts for the definition items that are returned by find all references for the case where "kind" and "name" are insufficient to properly classify the text. To support the new find all references APIs in Roslyn

### PR DESCRIPTION
**Pull request by riknoll**
Fixes #10541

Adds optional symbol display parts to the find all references results to aid in classifying the definition items that are returned.

@paulvanbrenk

### CODE REVIEW

File: src/services/services.ts [hunk 1]
```diff
@@ -1374,8 +1374,13 @@ namespace ts {
         containerName: string;
     }
 
+    export interface ReferencedSymbolDefinitionInfo extends DefinitionInfo {
+        // For more complex definitions where kind and name are insufficient to properly colorize the text
+        displayParts?: SymbolDisplayPart[];
```
Comment (mhegazy) [inline src/services/services.ts]: why is it optional? we are setting it all the time correct?

File: src/services/services.ts [hunk 2]
```diff
@@ -6634,14 +6640,20 @@ namespace ts {
                     getReferencesForStringLiteralInFile(sourceFile, type, possiblePositions, references);
                 }
 
+                const symbol = typeChecker.getSymbolAtLocation(node);
+
+                const displayParts = symbol ? getSymbolDisplayPartsDocumentationAndSymbolKind(
+                    symbol, node.getSourceFile(), getContainerNode(node), node).displayParts : undefined;
```
Comment (mhegazy) [inline src/services/services.ts]: Do not think you need the symbol here. you can always just add the string.

```ts
[displayPart(getTextOfNode(node), SymbolDisplayPartKind.stringLiteral)]
```

Comment (riknoll) [inline src/services/services.ts]: Yeah, I think you're right. I'll make the change. This also removes the case where display parts can be undefined, so I'll make that change too.

File: src/services/services.ts [hunk 3]
```diff
@@ -6641,7 +6646,8 @@ namespace ts {
                         fileName: node.getSourceFile().fileName,
                         kind: ScriptElementKind.variableElement,
                         name: type.text,
-                        textSpan: createTextSpanFromBounds(node.getStart(), node.getEnd())
+                        textSpan: createTextSpanFromBounds(node.getStart(), node.getEnd()),
+                        displayParts: [displayPart(getTextOfNode(node), SymbolDisplayPartKind.stringLiteral)]
```
Comment (paulvanbrenk) [inline src/services/services.ts]: Does this mean we always show the displayParts as stringLiterals?

**Comment by paulvanbrenk** [comment]
👍

**Comment by mhegazy** [comment]
we should add a fourslash test for it as well.

**Comment by mhegazy** [comment]
👍

**Comment by riknoll** [comment]
Added fourslash test and fixed a few errors

**Comment by mhegazy** [comment]
👍

**Pull request merged**